### PR TITLE
fixing algolia and init avails specs

### PR DESCRIPTION
**Pull request by fritzschoff**

### CODE REVIEW

File: libs/contract/src/lib/avails/avails.spec.ts [hunk 1]
```diff
@@ -0,0 +1,204 @@
+import { Term } from '../term/+state/term.model';
+import { isLicensed, filterSales } from './avails';
+import { mandateTerms as acTerms } from './fixtures/mandateTerms'
+import { saleTerms as acSaleTerms } from './fixtures/saleTerms';
+
+describe('Avails', () => {
+    let mandateTerms: Term<Date>[] = (acTerms as unknown[]).map(parseTimestampsOnTerms)
+    let saleTerms: Term<Date>[] = (acSaleTerms as unknown[]).map(parseTimestampsOnTerms)
+
+    it(`Mandate test (territory)
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/avails/avails.spec.ts:10]: Create a function to return the text based on params so you don't have to repeat yourself

File: libs/contract/src/lib/avails/avails.spec.ts [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        expect(ACRights).toBe(true);
+        const toDateSales = saleTerms.map(parseTimestampsOnTerms);
+        const avails = filterSales(
+            { duration: { to: new Date('06/30/2021'), from: new Date('01/01/2021') }, exclusive: true, territories: ['canada'], medias: ['freeTv'] },
+            toDateSales);
+        expect(avails).toBe(false);
+    });
+})
+
+function parseTimestampsOnTerms(term: Term<Date>): Term<Date> {
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/avails/avails.spec.ts:200]: can you rename it `toDate` ? It's easier to understand. Also the params should be `term: Term<Timestamp>` no ?

File: libs/contract/src/lib/avails/avails.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { Media, Territory } from "@blockframes/utils/static-model";
+import { Term } from "../term/+state/term.model";
+
+interface AvailsFilter { medias: Media[], duration: { from: Date, to: Date }, territories: Territory[], exclusive: boolean }
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/avails/avails.ts:4]: format the interface on several lines please.

File: libs/contract/src/lib/term/+state/term.model.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { MovieLanguageSpecification } from "@blockframes/movie/+state/movie.firestore";
 
-export interface Term {
+export interface Term<T> {
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/term/+state/term.model.ts:3]: If you want you can do : `Terms<T extends Date | Timestamp = Date>`. Like that we only need to specify if it's Timestamp.

File: libs/firebase-utils/src/lib/algolia/algolia.ts [hunk 5]
```diff
@@ -37,13 +37,15 @@ export function clearIndex(indexName: string, adminKey?: string) {
...
 
-export function setIndexConfiguration(indexName: string, config: AlgoliaConfig, adminKey?: string) {
+export function  setIndexConfiguration(indexName: string, config: AlgoliaConfig, adminKey?: string) {
   if (!algolia.adminKey && !adminKey) {
     console.warn('No algolia id set, assuming dev config: skipping');
     return Promise.resolve(true);
   }
 
-  return indexBuilder(indexName, adminKey).setSettings(config);
+  return indexBuilder(indexName, adminKey).setSettings({
```
Comment (GrandSchtroumpf) [inline libs/firebase-utils/src/lib/algolia/algolia.ts:46]: format on several lines for readability please.

File: package.json [hunk 6]
```diff
@@ -5,6 +5,7 @@
   "scripts": {
     "ng": "nx",
     "start": "nx serve --hmr --disable-host-check",
+    "start:catalog": "nx serve catalog --hmr",
```
Comment (GrandSchtroumpf) [inline package.json:8]: Do we really need to add that ? I feel like package.json is already too fat ^^

**Review by GrandSchtroumpf — APPROVED** [review]

File: package.json (hunk 6 above)
Comment (fritzschoff) [reply package.json:8]: no fat shaming please ;)

**Pull request merged**